### PR TITLE
Implemented API to test s3 connection.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
Implemented API to test s3 connection.

This PR is dependent on PR https://github.com/opensearch-project/security-analytics-commons/pull/11
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java [hunk 1]
```diff
@@ -219,6 +222,8 @@ public class SecurityAnalyticsPlugin extends Plugin implements ActionPlugin, Map
     public static final String THREAT_INTEL_SOURCE_URI = PLUGINS_BASE_URI + "/threat_intel/source";
     public static final String THREAT_INTEL_MONITOR_URI = PLUGINS_BASE_URI + "/threat_intel/monitor";
     public static final String LIST_IOCS_URI = PLUGINS_BASE_URI + "/iocs/list";
+    public static final String TEST_CONNECTION_BASE_URI = PLUGINS_BASE_URI + "/test_connection";
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java:225]: ideally api should be `/_plugins/_sap/connections/test/s3`

File: src/main/java/org/opensearch/securityanalytics/action/TestS3ConnectionRequest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.core.xcontent.XContentParserUtils;
+import org.opensearch.securityanalytics.commons.connector.model.S3ConnectorConfig;
+
+import java.io.IOException;
+
+public class TestS3ConnectionRequest extends ActionRequest implements ToXContentObject {
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/action/TestS3ConnectionRequest.java:21]: why si this object not like ?
```
class TestS3ConnectionRequest
     S3Config s3config
}
```

plz reuse whatever is present in the threat intel source config

File: src/main/java/org/opensearch/securityanalytics/transport/TransportTestS3ConnectionAction.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    private final STIX2IOCFetchService stix2IOCFetchService;
+
+    @Inject
+    public TransportTestS3ConnectionAction(
+            TransportService transportService,
+            ActionFilters actionFilters,
+            STIX2IOCFetchService stix2IOCFetchService
+            ) {
+        super(TestS3ConnectionAction.NAME, transportService, actionFilters, TestS3ConnectionRequest::new);
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportTestS3ConnectionAction.java:36]: we need S3Exception.awsErrorDetails() also?

File: src/main/plugin-metadata/plugin-security.policy [hunk 4]
```diff
@@ -1,8 +1,26 @@
...
+
 grant {
     // needed to find the classloader to load whitelisted classes.
     permission java.lang.RuntimePermission "createClassLoader";
     permission java.lang.RuntimePermission "getClassLoader";
 
     permission java.net.SocketPermission "*", "connect,resolve";
     permission java.net.NetPermission "getProxySelector";
+
+    permission java.io.FilePermission "${user.home}${/}.aws${/}*", "read";
```
Comment (eirsep) [inline src/main/plugin-metadata/plugin-security.policy:17]: are these still needed?

**Review by eirsep — CHANGES_REQUESTED** [review]
some comments

File: src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java (hunk 1 above)
Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java:225]: Revised.

File: src/main/java/org/opensearch/securityanalytics/transport/TransportTestS3ConnectionAction.java (hunk 3 above)
Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportTestS3ConnectionAction.java:36]: Could you clarify what you mean by this? I did some searching in the OpenSearch project, but didn't come across `awsErrorDetails` used in transport action constructors.

File: src/main/java/org/opensearch/securityanalytics/action/TestS3ConnectionRequest.java (hunk 2 above)
Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/action/TestS3ConnectionRequest.java:21]: Revised.

File: src/main/java/org/opensearch/securityanalytics/transport/TransportTestS3ConnectionAction.java (hunk 3 above)
Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportTestS3ConnectionAction.java:36]: Clarified offline that we should try to use the `awsErrorDetails` to provide the error message in the `doExecute` function responses.

Revising accordingly.

File: src/main/plugin-metadata/plugin-security.policy (hunk 4 above)
Comment (AWSHurneyt) [reply src/main/plugin-metadata/plugin-security.policy:17]: My local testing suggests we only need to add the `permission java.io.FilePermission "${user.home}${/}.aws${/}*", "read";` permission. I've removed the others.

File: src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java (hunk 1 above)
Comment (eirsep) [reply src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java]: apologies.. /_plugins/_sap/connections/s3/test

Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java]: Revised.

**Review by eirsep — APPROVED** [review]

**Review by jowg-amazon — APPROVED** [review]

**Pull request merged**